drivers/SmartThings/matter-switch/src/aqara-cube/init.lua

```diff
@@ -0,0 +1,217 @@
+      local flag = 0
+      local previous = 0
+      for _, ep in ipairs(MS) do
+        if idx1 == 1 then
+          ept1[idx1] = ep
+        else
+          if flag == 0
+            and ep <= (previous + 15) then
+            -- the endpoint number does not always increase by 1
+            -- as a workaround, assume that the next endpoint number is not greater than 15
+            ept1[idx1] = ep
+          else
+            ept2[idx2] = ep
+            idx2 = idx2 + 1
+            if flag ~= 1 then
+              flag = 1
+            end
+          end
+        end
+        idx1 = idx1 + 1
+        previous = ep
+      end
+
+      local start = #ept2 + 1
+      idx1 = 1
+      idx2 = start
+      for i=start, 6 do
+        ept2[idx2] = ept1[idx1]
+        idx1 = idx1 + 1
+        idx2 = idx2 + 1
+      end
+      return ept2
+    end
+  end
+end
+
+local function endpoint_to_component(device, endpoint)
+  return "main"
+end
+
+local function device_init(driver, device)
+  if device.network_type == device_lib.NETWORK_TYPE_MATTER then
+    device:subscribe()
+    device:set_endpoint_to_component_fn(endpoint_to_component)
+  end
+end
+
+-- This is called either on add for parent/child devices, or after the device profile changes for components
+local function configure_buttons(device)
+  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+    local MS = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH})
+    device.log.debug(#MS.." momentary switch endpoints")
+    for _, ep in ipairs(MS) do
+      -- device only supports momentary switch, no release events
+      device.log.debug("configuring for press event only")
+      set_field_for_endpoint(device, INITIAL_PRESS_ONLY, ep, true, true)
+    end
+  end
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+    local MS = get_reordered_endpoints(driver, device)
+    local main_endpoint = device.MATTER_DEFAULT_ENDPOINT
+    if #MS > 0 then
+      main_endpoint = MS[1] -- the endpoint matching to the non-child device
+      if MS[1] == 0 then main_endpoint = MS[2] end -- we shouldn't hit this, but just in case
+    end
+    device.log.debug("main button endpoint is "..main_endpoint)
+
+    -- At the moment, we're taking it for granted that all momentary switches only have 2 positions
+    local current_component_number = 1
+    local component_map = {}
+    for _, ep in ipairs(MS) do -- for each momentary switch endpoint (including main)
+      device.log.debug("Configuring endpoint "..ep)
+      -- build the mapping of endpoints to components
+      component_map[string.format("%d", current_component_number)] = ep
+      current_component_number = current_component_number + 1
+    end
+
+    device:set_field(COMPONENT_TO_ENDPOINT_MAP, component_map, {persist = true})
+    device:try_update_metadata({profile = "cube-t1-pro"})
+    configure_buttons(device)
```

Comment:
If we are updating the profile name and deferring configuration of the buttons until the profile is updated, then shouldn't `configure_buttons` not be called here? In the main driver, if a new profile is going to be applied, `configure_buttons` is not called until `info_changed` is called after the profile update.

Reply:
When the hub is restarted, `info_changed` is not called and `configure_buttons` cannot be called, so it is also added here. Is there anything I checked wrong?

Reply:
My understanding is that `configure_buttons` only needs to be called the first time a device is initialized and `info_changed` is called after the profile is joined, so I don't understand the need to call it in `device_added`. Is there a reason it would need to be called when the hub restarts?

Reply:
I checked again and it seems to be a little different from matter-button's lifecycle event. In conclusion, you were right, so I deleted the `configure_buttons` call from `device_added`. Thank you very much~